__init__.py

```diff
@@ -1150,6 +1151,43 @@ def _speak_weather(self, dialog):
         self.log.info("Speaking dialog: " + dialog.name)
         self.speak_dialog(dialog.name, dialog.data, wait=True)
 
+    @skill_api_method
+    def get_current_local_weather(self):
+        """Get the current temperature and weather condition.
```

Comment:
This docstring could be improved (e.g. "Expose current weather conditions to other skills via the skill API").  Also, do we really need to define the dictionary in the docstring and the code?  Seems a bit superfluous.

Reply:
I think if we do this for each Skill API method then every single one would read: `f"Expose {description} to other Skills via the Skill API"`. To me this is still implied by the decorator.

I guess a concern is that we don't want to let zombie code live in Skills so we want to make it clear why some function exists. So whilst the Skill itself could use this method if it wanted to, it's not, and is unlikely to in the future.

Do we need this detail in the docstring or just in the code?
The docstring will appear in the Skill API listing, and in IDE's. The API listing doesn't need that detail because it's what the list is all about. Does the IDE really need it?
Eg We could add a sort of section break in the code:
```
#####################################
######### SKILL API METHODS #########
##################################### 
```
So someone reading the code itself can see why it's there, but we don't clutter the Skill API listing with repetitive text.

Reply:
In terms of the dictionary in the docstring, my thinking was that it provided Skill authors using the method via the Skill API all the info they needed. Otherwise you'd need to first call it, then print out the keys you're getting returned and make assumptions about what they represent.

If you haven't seen it already, with this branch checked out, in the CLI type:
```
:api mycroft-weather.mycroftai
```